strapi/strapi
# ci: fix docs workflow jobs

### What does it do?

- removes the `skipped_tests.yml` which duplicated+mimicked the real workflow and instead rely on conditionals in `tests.yml`
- fixes the issue that docs-only workflows do not ever complete the aggregate
- now tests docs build process instead of docs-only changes always passing
- unify test conditionals

TODO in a followup PR:

- combine install+build and possibly conditionals into a single action (see https://github.com/strapi/strapi/pull/21783 ) which should reduce the amount of code needed
- find a way to split this up into multiple files without complicating it too much; the issue is that workflows can only communicate through artifacts, so the amount of setup and output necessary in each file might make it not worth it unless they can somehow rely on something like a single action to handle everything

### Why is it needed?

Currently without this change:

- We had to set up a duplicate 'aggregate' in the docs-only workflow, and in order for it to still show "skipped" in github we had to duplicate all existing jobs
- because of that, it regularly got out-of-sync or broken
- When we have a docs-only PR, aggregate doesn't run and tests never pass/fail, they just stall
- When we have PRs that mix docs changes and code changes, there is ambiguous behaviour since both workflows run with identical job names inside
- We never test that docs can build after docs changes
- Our conditional statements are long and confusing

### How to test it?

Create PRs that want to merge into this branch with the following cases of changed files:

docs: should only run docs_build 
front-end: should only run front-end and e2e tests
back-end: should only run back-end, e2e, typescript, cli
github workflow, package.json, jest file: should run all tests

in all cases, aggregate_test_result should pass once the run is complete and build, lint, and pretty should still run


### Related issue(s)/PR(s)

Let us know if this is related to any issue/pull request
